pkg/sdk/roles_validations.go

```diff
@@ -0,0 +1,93 @@
+package sdk
+
+import "errors"
+
+func (opts *CreateRoleOptions) validate() error {
+	if opts == nil {
+		return ErrNilOptions
```

Comment:
I was returning for such case `errors.Join(errNilOptions)`, so that we always return the "same" thing, but I guess that output should be really similar in this case

Reply:
Yeah, It doesn't matter in this case. The only thing is different is if you want to assert on error with wrapping you're asserting stringError inside joinedError and this is only stringError. I think we can leave it as is, because it may be changed after we'll figure out how do we want handle errors with new custom error type.